src/JsonSchema/Constraints/Constraint.php

```diff
@@ -242,7 +261,7 @@ protected function checkEnum($value, $schema = null, $path = null, $i = null)
 
     protected function checkFormat($value, $schema = null, $path = null, $i = null)
     {
-        $validator = new FormatConstraint($this->checkMode, $this->uriRetriever);
+        $validator = $this->getFactory()->factory('format');
         $validator->check($value, $schema, $path, $i);
 
         $this->addErrors($validator->getErrors());
```

Comment:
As a thought - some of the duplication of this class could now be moved into a private method. ```checkObject``` would need to remain as it is - as it also passes ```$patternProperties```

```php
    protected function checkFormat($value, $schema = null, $path = null, $i = null)
    {
        $this->checkByType('format', $value, $schema, $path, $i);
    }

    /**
     * Create a validator of type $type and validate value against it
     * 
     * @param string $type
     * @param mixed  $value
     * @param mixed  $schema
     * @param mixed  $path
     * @param mixed  $i
     */
    private function checkByType($type, $value, $schema, $path, $i)
    {
        $validator = $this->getFactory()->factory($type);
        $validator->check($value, $schema, $path, $i);

        $this->addErrors($validator->getErrors());
    }
```